README.md

```diff
@@ -1,347 +1,516 @@
 
-```
-/*************************/
-Ancestor Program: [Path to the program including the name of the program]
-original data: [Path of where the data live]
-Description: [Overall description]
-(1)[insert task description here, and then copy & paste this to indicate where that task is later in your program]
-(2)
-(3) [etc...]
-*/
-/*************************/
+#### Assertions and Tests
 
-```
+Write assertions and in-line tests.
+Assertions, things expected to always be true about the code, should be tested in-line.
+[healthinequality-code](https://github.com/michaelstepner/healthinequality-code/blob/master/code/readme.md#assert-what-youre-expecting-to-be-true) offers some good background.
+`assert` is useful in Stata and `stopifnot()` is useful in R.
 
-## Subgroups Code
+Write tests for final files.
+For example, write a test if all numbers should be non-negative or if values should not exceed \$3,000.
 
-Metric leads will need to decide whether to create new scripts/programs for extending the database (additional years or subgroup analysis) or to extend existing scripts. The optimal approach may differ based on the situation. For example, some metric leads will need to change data sets entirely (e.g. 1-year vs. 5-year ACS data) and new scripts may be most efficient and clean, while other metric leads may need to make minimal changes to an existing script.  
+Check that the value ranges make sense.
+Spot-check your outliers to confirm that those values are not an error.
+In some cases, you might need to dig on the internet to see if say a community has the worst rates of air quality that there's some verification of that.
+Or if homelessness among students spiked that there's some context that could explain that.
+If not, then check your code to make sure it is doing what you think it is doing.
 
-1. Do not worry about editing existing scripts provided they recreate the original dataset. There is no need to use version control in file names (i.e. script.R to script2.R). Git will handle this. If you have not added your data as a .csv to version control, then please reach out to Aaron (this is a change from earlier policy). 
-2. When possible, use functions and macros to avoid repeating the same code multiple times. For example, if you are pulling the 1-year ACS, try to write functions that take year as an argument and then call the function multiple times instead of copying-and-pasting code.  
+Also, check the data quality flag.
+Look at the distribution of assigned quality.
+
+Write tests for macros and functions to ensure appropriate behavior.
+
+> Whenever you are tempted to type something into a print statement or a debugger expression, write it as a test instead.
+> --- Martin Fowler
+
+#### Benchmarking
+
+Check final calculations against state and/or national numbers if available.
+
+For questions about code, please contact [upwardmobility\@urban.org](mailto:upwardmobility@urban.org) and include "Code in GitHub" in the subject line.
```

Comment:
This seems out of place.

Reply:
Claudia suggested this as a way to formalize the contact process, I'm indifferent